modules/webclient/src/main/scala/scaladex/client/ActiveNavObserver.scala

```diff
@@ -11,20 +13,32 @@ import scaladex.dom.IntersectionObserver
   * list items in nav.
   */
 object ActiveNavObserver {
-  private val observer: IntersectionObserver = IntersectionObserver { entry =>
-    val id = entry.target.getAttribute("id")
-    val navLink = document.querySelector(s"""nav li a[href="#$id"]""").asInstanceOf[Link]
-    if (navLink != null) {
-      val navLI = navLink.parentElement.asInstanceOf[LI]
+  def start(): Unit = {
+    val sectionsAndNavItem =
+      document
+        .querySelectorAll("section[id]")
+        .toSeq
+        .collect { case e: Element => e }
+        .flatMap { section =>
+          val id = section.getAttribute("id")
+          val link = document.querySelector(s"""nav li a[href="#$id"]""").asInstanceOf[Link]
+          Option(link).map(l => section -> l.parentElement)
+        }
 
-      if (entry.intersectionRatio > 0) navLI.classList.add("active")
-      else navLI.classList.remove("active")
+    val observer = IntersectionObserver { _ =>
+      // ignore the observed entry and update all sections instead
+      timers.setTimeout(0) {
```

Comment:
So I looked at the solution at https://stackoverflow.com/a/65008310/1829647 and now I understand better why they use a `setTimeout`. They actually use it in conjunction with a `clearTimeout`, with the same handle `debounceTimeout`. With the 150 milliseconds, this effectively achieves a *debounce* mechanism: if the intersection observer fires multiple times within 150 ms, the recomputation will only happen once, after it has "stabilized" for at least 150 ms. I think it would be good to take this mechanism and implement here.

You'll have to declare `debounceTimeout` as a
```scala
var debounceTimeout: js.timers.SetTimeoutHandle = null
```